scripts/globals/magic.lua

```diff
@@ -618,211 +467,124 @@ function applyResistanceEffect(player,spell,target,diff,skill,bonus,effect)
-
-    --add a flat bonus that won't get doubled in the previous step
-    p = p + 45;
-
-    --add a scaling bonus or penalty based on difference of targets level from caster
-    local leveldiff = player:getMainLvl() - target:getMainLvl();
-    if (leveldiff < 0) then
-        p = p - (25 * ( (player:getMainLvl()) / 75 )) + leveldiff;
-    else
-        p = p + (25 * ( (player:getMainLvl()) / 75 )) + leveldiff;
-    end
-    --cap accuracy
-    if (p > 95) then
-        p = 95;
-    elseif (p < 5) then
-        p = 5;
-    end
-
-    p = p / 100;
-
-    -- Resistance thresholds based on p.  A higher p leads to lower resist rates, and a lower p leads to higher resist rates.
-    local half = (1 - p);
-    local quart = ((1 - p)^2);
-    local eighth = ((1 - p)^3);
-    local sixteenth = ((1 - p)^4);
-    -- print("HALF:",half);
-    -- print("QUART:",quart);
-    -- print("EIGHTH:",eighth);
-    -- print("SIXTEENTH:",sixteenth);
-
-    local resvar = math.random();
-
-    -- Determine final resist based on which thresholds have been crossed.
-    if (resvar <= sixteenth) then
-        resist = 0.0625;
-        --printf("Spell resisted to 1/16!!!  Threshold = %u",sixteenth);
-    elseif (resvar <= eighth) then
-        resist = 0.125;
-        --printf("Spell resisted to 1/8!  Threshold = %u",eighth);
-    elseif (resvar <= quart) then
-        resist = 0.25;
-        --printf("Spell resisted to 1/4.  Threshold = %u",quart);
-    elseif (resvar <= half) then
-        resist = 0.5;
-        --printf("Spell resisted to 1/2.  Threshold = %u",half);
-    else
-        resist = 1.0;
-        --printf("1.0");
-    end
-
-    return resist;
-
+    return getMagicResist(p);
 end;
 
 --Applies resistance for additional effects
 function applyResistanceAddEffect(player,target,element,bonus)
 
-    local resist = 1.0;
-    local magicaccbonus = 0;
-
-    if (bonus ~= nil) then
-        magicaccbonus = magicaccbonus + bonus;
-    end
-
-    --get the base acc (just skill plus magic acc mod)
-    local magicacc = 0;
+    -- Add a base 75 magic acc
+    local p = getMagicHitRate(player, target, 0, element, 75, bonus);
 
-    --add acc for staves
-    local affinityBonus = AffinityBonus(player, element);
-    magicaccbonus = magicaccbonus + (affinityBonus-1) * 200;
+    return getMagicResist(p);
+end;
 
-    --base magic evasion (base magic evasion plus resistances(players), plus elemental defense(mobs)
-    local magiceva = target:getMod(resistMod[element]);
+function getMagicHitRate(caster, target, skillType, element, fullBonus, minorBonus)
+  -- resist everything if magic shield is active
+  if (target:hasStatusEffect(EFFECT_MAGIC_SHIELD, 0)) then
+    return 0;
+  end
+
+  local magicacc = 0;
+  local magiceva = 0;
+  local p = 0;
+
+  if(minorBonus == nil) then
+    minorBonus = 0;
+  end
+
+  -- Get the base acc (just skill + skill mod (79 + skillID = ModID) + magic acc mod)
+  local magicacc = caster:getMod(MOD_MACC);
+
+  if(skillType ~= 0) then

```

Comment:
stick a space after between if/elseif's and ('s cause readability
my second monitor sucks :(